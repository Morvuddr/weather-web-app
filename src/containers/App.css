.App-header {
  display: flex;
  flex-direction: column;
  height: 160px;
  width: 90%;
  margin: auto;
}
.Refresh-Location-Button {
  width: 200px;
  height: 40px;
  border-radius: 5px;
  font-size: 12px;
  background: lightgrey;
}

@media (min-width: 700px) {
  
  .App-header{
    
    position: relative;
    flex-direction: row;
    height: 80px;

  }
  .Refresh-Location-Button {
    position: absolute;
    top: 50%;
    margin-top: -20px;
    left: 50%;
    margin-left: -100px;
    
  }
}