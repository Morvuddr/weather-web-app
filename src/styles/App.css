.App-header {
  display: flex;
  flex-direction: column;
  height: 160px;
  width: 90%;
  margin: auto;
}
.Refresh-Location-Button {
  width: 200px;
  height: 40px;
  border-radius: 5px;
  font-size: 12px;
  background: lightgrey;
}

@media (min-width: 700px) {
  
  .App-header{
    
    position: relative;
    flex-direction: row;
    height: 80px;

  }
  .Refresh-Location-Button {
    position: absolute;
    top: 50%;
    margin-top: -20px;
    left: 50%;
    margin-left: -100px;
    
  }
}

.Loader{
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}
.Loader h1{
  text-align: center;
}
.Spinner{
  border: 16px solid #f3f3f3;
  border-top: 16px solid gray;
  border-radius: 50%;
  width: 60px;
  height: 60px;
  animation: spin 2s linear infinite;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}
.WeatherInfo{
  width: 45%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.Info{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 40px;
  background: gray;
}
.Info h3{
  margin: 0px;
  padding-left: 10px;
  color: white;
}
.Info h4{
  margin: 0px;
  padding-right: 10px;
  color: lightgray;
}
.Error {
  text-align: center;
}