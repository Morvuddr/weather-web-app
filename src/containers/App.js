import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';
import MainContainer from './MainContainer';
import LocationService from '../services/LocationService';
import Header from '../components/Header';
import FavoritesContainer from './FavoritesContainer';
import { initCities } from '../redux/favorites/actions';

class App extends Component {
    state = {
        isLoading: true,
        weather: {},
    };

    constructor(props) {
        super(props);
        this.initWeather();
        props.initCities();
    }

    initWeather = () => {
        LocationService.handleUpdateLocation()
            .then(weather => this.setState({ weather, isLoading: false }));
    };

    handleUpdateWeather = () => {
        this.setState({
            isLoading: true
        });

        LocationService.handleUpdateLocation()
            .then(weather => this.setState({ weather, isLoading: false }));
    };

    render() {
        console.log(this.props.city);
        const { isLoading, weather } = this.state;
        return (
            <div className='App'>
                <Header onUpdate={this.handleUpdateWeather}/>
                <MainContainer weather={weather} isLoading={isLoading}/>
                <FavoritesContainer cities={this.props.cities}/>
            </div>
        );
    }
}

const mapStateToProps = ({ city }) => {
    return {
        city: city,
    };
};
const mapDispatchToProps = {
    initCities: initCities
};

export default connect(mapStateToProps, mapDispatchToProps)(App);

