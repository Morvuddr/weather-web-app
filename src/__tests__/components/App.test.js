import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../../containers/App';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('App container', () => {

    test('matches the snapshot when local weather is loaded and favorites cities are loaded', () => {
        const city = {
            isLoading: false,
            error: false,
            weather: {
                name: 'Moscow',
                img: 'abcd',
                temperature: 'temp',
                wind: 'wind',
                cloudiness: 'cloudiness',
                pressure: 'pressure',
                humidity: 'humidity',
                location: '[location]'
            }
        };
        const cities = [{
            id: '1',
            name: 'Moscow',
            img: 'abcd',
            temperature: 'temp',
            wind: 'wind',
            cloudiness: 'cloudiness',
            pressure: 'pressure',
            humidity: 'humidity',
            location: '[location]'
        }];
        const store = mockStore({
            city: city,
            favorites: {
                cities: cities,
                errors: [],
                isLoading: []
            }
        });
        const tree = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});