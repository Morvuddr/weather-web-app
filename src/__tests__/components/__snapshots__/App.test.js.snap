// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container matches the snapshot when local weather is loaded and favorites cities are loaded 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Погода здесь
    </h1>
    <button
      className="Refresh-Location-Button"
      onClick={[Function]}
    >
      Обновить геолокацию
    </button>
  </header>
  <div
    className="Main"
  >
    <div
      className="currentCity"
    >
      <h1>
        Moscow
      </h1>
      <div
        className="additionalInfo"
      >
        <img
          alt="Иконка погоды"
          className="weatherImage"
          src="abcd"
        />
        <h1>
          temp
        </h1>
      </div>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          wind
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          cloudiness
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          pressure
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          humidity
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          [location]
        </h4>
      </div>
    </div>
  </div>
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      >
        +
      </button>
    </form>
  </div>
  <div
    className="FavList"
  >
    <div
      className="FavoriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="abcd"
        />
        <h1>
          temp
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          ×
        </button>
      </div>
      <div
        className="WeatherInfo"
      >
        <div
          className="Info"
        >
          <h3>
            Ветер
          </h3>
          <h4>
            wind
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Облачность
          </h3>
          <h4>
            cloudiness
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Давление
          </h3>
          <h4>
            pressure
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Влажность
          </h3>
          <h4>
            humidity
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Координаты
          </h3>
          <h4>
            [location]
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
